### DISCUSSION TITLE
Conditional Database Seeders

### DISCUSSION THREAD

**Discussion by rashidlaasri** [laravel/framework#52708] [Ideas]
I am currently working with multiple Database Seeder classes and some of those classes should only run locally, so, I was thinking rather than having to wrap them in an if condition, maybe we could introduce a better API/Syntax?

- Proposition 1: Add a `shouldRun` (or similar name) method to the seeder class
```php
public function shouldRun(): bool
{
    return app()->environment('local');
}
```

- Proposition 2: Chain on the `$this->call()` 
```php
$this->call([
    UsersSeeder::class,
])->local();
```

**Comment by shaedrich** [comment]
There could even be a reusable attribute like `#[Environment('local')]`

**Comment by josephkerkhof** [comment]
Pull request to implement this idea: https://github.com/laravel/framework/pull/57399
